src/client/Client.cc

```diff
@@ -2663,21 +2663,31 @@ void Client::handle_mds_map(MMDSMap* m)
 
     int oldstate = oldmap->get_state(mds);
     int newstate = mdsmap->get_state(mds);
-    if (!mdsmap->is_up(mds)) {
+    if (!mdsmap->is_up(mds) || mdsmap->get_addrs(mds) != session->addrs) {
       session->con->mark_down();
-    } else if (mdsmap->get_addrs(mds) != session->addrs) {
-      session->con->mark_down();
-      session->addrs = mdsmap->get_addrs(mds);
-      // When new MDS starts to take over, notify kernel to trim unused entries
-      // in its dcache/icache. Hopefully, the kernel will release some unused
-      // inodes before the new MDS enters reconnect state.
-      trim_cache_for_reconnect(session);
+      if (mdsmap->get_addrs(mds) != session->addrs) {
```

Comment:
if mds it not up, mdsmap->get_addrs(mds) will cause crash

Reply:
i will change it later.